src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import java.util.Scanner;
+
+import exception.DukeException;
+import parser.Parser;
+import storage.Storage;
+import task.Task;
+import tasklist.TaskList;
+import ui.Ui;
+
+
+/**
+ * Represents the whole program, containing the main program for the whole bot.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for a Duke Object.
+     * Initializes the whole program as well as required objects.
+     * If path leads to a file that does not exist, the file will be created.
+     *
+     * @param filePath String of path to stored list from previous sessions.
+     * @return Duke object.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+        }
+    }
+
+    /**
+     * Function that runs while waiting for user input.
+     */
+    public void run() {
+        String echo;
+        Scanner scan = new Scanner(System.in);
+
+        while (true) {
+            echo = scan.nextLine();
+
+            if (echo.equals("bye")) {
+                break;
+            }
+
+            if (Parser.parseCommands(echo, tasks, ui)) {
+                continue;
+            }
+
+            Task item;
+
+            // Create a separate function in order to assign to item;
+            try {
+                item = Parser.parseEcho(echo);
+            } catch (DukeException e) {
+                // TODO: handle exception
+                System.out.println(e.getMessage());
+                continue;
+            }
+
+            tasks.addTask(item);
+
+            ui.showAddedMessage(item);
+            ui.printListNumber(tasks.getList());
+        }
+
+        ui.showSavingMessage();
+
+        storage.save(tasks.getList());
+
+
+        ui.showSavedMessage();
+
+        scan.close();
+
+        ui.showClosingMessage();
```

Comment:
Can group these lines of code together and remove the whitespace since they are all related to the task of closing the application.